java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -347,27 +431,45 @@ protected final <Z> Z createProxy(final Decorated<Z> decorated, Class<Z> clazz)
     allInterfaces.addAll(originalInterfaces);
     allInterfaces.addAll(derivedInterfaces.keySet());
     Class<?>[] allInterfacesArray = allInterfaces.toArray(new Class<?>[0]);
+    // ensure we do not hit an existing field
+    int fieldName;
+
+    for (fieldName = 0; fieldName < 8192; fieldName++) {
+      try {
+        clazz.getDeclaredField("___target" + fieldName);
+      } catch (NoSuchFieldException ex) {
+        break;
+      }
+    }
+
+    if (fieldName == 8192) {
+      throw new AssertionError("No free field name found");
+    }
 
     Class<? extends Z> proxy =
         new ByteBuddy()
             .subclass(clazz.isInterface() ? Object.class : clazz)
             .implement(allInterfacesArray)
             .method(ElementMatchers.any())
             .intercept(InvocationHandlerAdapter.of(handler))
+            .defineField("___target" + fieldName, sample.getClass(), Visibility.PUBLIC)
```

Comment:
see above. I would rather prefer having this field private under an interface with setter and getter